### PR TITLE
Datatypes

### PR DESCRIPTION
**Pull request by psc478**
## Checkpoint Rubric
This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

**Checkpoint 1**
- [ ] All tests passed: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 30 points

**Checkpoint 2**
- [ ] The application works as it should: 40 points
- [ ] Proper use of documentation (commenting on code): 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript and the DOM API: 30 points

**Checkpoint 3**
- [ ] Use of React: 25 points
- [ ] Accesses an API: 25 points
- [ ] Proper use of documentation (commenting on code): 25 points
- [ ] The application functions as it should: 25 points

### CODE REVIEW

**Review by yousifS — APPROVED** [review]
Nice

File: 01week/datatypes.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+//convert string to number
+let stringToNumber = function(y){
+  if(typeof(y)==='string'){
+    y = Number(y);
+    console.log("3. " + y + " is now a number. typeof(" + y + ")=" + typeof(y));
+  }
+}
+
+//display datatype of input
+let whatDataType = function(z){
```
Comment (yousifS) [inline 01week/datatypes.js:25]: What will happen when you run the following examples? what would you expect the answer to be? 
```
whatDataType(NaN);
whatDataType(null);
```

Comment (psc478) [reply 01week/datatypes.js:25]: I expect the answer to be:

whatDataType(NaN):
NaN has a datatype of number.

whatDataType(null):
null has a datatype of object.

...I ran the program, and verified that I am correct.

**Comment by yousifS** [comment]
The assignment is asking you to distinguish between

   - Boolean
   - Null
   - Undefined
   - Number
   - NaN
   - String


Even though typeof(null) is object, and typeof(NaN) is number.

**Pull request merged**